### PR TITLE
Clearing up the subinst

### PR DESCRIPTION
**Pull request by mkosunen**
Comments on _subckt

Disabling re-parsing of subinst from subckt_file as the string has been extracted alrethe string .

### CODE REVIEW

**Comment by mkosunen** [comment]
The main problems causing confusion:
 Questions:

1. In pre layout, the file is NOT copied as it may contain something else than the subckt definitions (i.e simulator directives)
In post layout it MUST be copied because of what? Slow parsing? I'll leave it for now. In any case, 'subinst' should not parse the file again. 


2. In some cases the name of the subcircuit is detected based on the string determined by the extraction program, and changed in place to something else. Why? What is the reason for this? What causes the discrepancy between the entity name and the subcircuit name?

**Comment by sporrasm** [comment]
My two cents (don't have solid info on this since I haven't originally written any of this).

1. In prelayout, the file is indeed not copied, as it needs to be pruned of extra options as you said. In postlayout there are two cases for parsing the subckt. Calibre formatted and DSPF formatted netlists. In case of Calibre, the comments seem to suggest that the file is indeed copied to the testbench directory and modified in place due to the speed of the operation. Reading in postlayout netlists, and editing in Python can be rather slow as the netlists are often quite huge. For DSPF formatted netlists, the subckt is parsed based on the prelayout netlist and the postlayout netlist is included in to the testbench file (tb_entityname.postfix).  

2. The discrepancy between the entity name and the subcircuit name is caused by the fact that the netlist top cell is named after the cell it was generated from. E.g. if I netlist a design 'filter_lowpass', the top cell name will be 'filter_lowpass'. As to why it must be changed, I've no idea. I'm not sure if that is strictly necessary. Leaving the renaming out and using the pre-generated top cell name in the testbench could be tried as well. I guess this would save some time, replacing the cellname from the netlist seems to take some time.

**Comment by sporrasm** [comment]
I tested this with Calibre and DSPF formatted netlists. Seems to be working just fine.

**Review by sporrasm — APPROVED** [review]

File: spice/spice_module.py [hunk 1]
```diff
@@ -115,31 +115,44 @@ def subckt(self):
...
                                 # Copying the file and editing it seems better
-                                self.postlayout = True
+
                                 # Right now this just ignores everything and overwrites the subcircuit file
                                 # TODO: think about this
                                 if os.path.isfile(self._subcktfile):
                                     os.remove(self._subcktfile)
                                     shutil.copyfile(self._dutfile,self._subcktfile)
                                 else:
                                     shutil.copyfile(self._dutfile,self._subcktfile)
```
Comment (sporrasm) [inline spice/spice_module.py:136]: Just a quick thought: I don't really understand we need a separate self._dutfile property. This is set in testbench.py and just points to parent's self.spicesrc, the netlist source.  Why confuse others with obscure variable names that do not really describe what they contain/are used for?

This is more likely related to general cleanup and can be done in #52. The whole code base is full of these obscure variables which should be cleaned up in my opinion. What say you? @mkosunen

**Pull request merged**